core/server/api_container/server/startosis_engine/startosis_packages/git_package_content_provider/git_package_content_provider_test.go

```diff
@@ -211,10 +236,37 @@ func TestGitPackageProvider_FailsForNonExistentPackage(t *testing.T) {
 	provider := NewGitPackageContentProvider(oackageDir, packageTmpDir, githubAuthTokenFilePath.Name(), nil)
 	nonExistentModulePath := "github.com/kurtosis-tech/non-existent-startosis-load/sample.star"
 
-	_, err = provider.GetModuleContents(nonExistentModulePath)
+	nonExistentModuleAbsoluteLocator := startosis_packages.NewPackageAbsoluteLocator(nonExistentModulePath, defaultMainBranch)
+
+	_, err = provider.GetModuleContents(nonExistentModuleAbsoluteLocator)
 	require.NotNil(t, err)
 }
 
+func TestGitPackageProvider_GetContentFromAbsoluteLocatorWithCommit(t *testing.T) {
+	packageDir, err := os.MkdirTemp("", packagesDirRelPath)
+	require.Nil(t, err)
+	defer os.RemoveAll(packageDir)
+	packageTmpDir, err := os.MkdirTemp("", repositoriesTmpDirRelPath)
+	require.Nil(t, err)
+	defer os.RemoveAll(packageTmpDir)
+	githubAuthDir, err := os.MkdirTemp("", githubAuthDirRelPath)
+	require.Nil(t, err)
+	githubAuthTokenFilePath, err := os.CreateTemp(githubAuthDir, githubAuthTokenFilename)
+	require.Nil(t, err)
+	defer os.RemoveAll(githubAuthDir)
+
+	provider := NewGitPackageContentProvider(packageDir, packageTmpDir, githubAuthTokenFilePath.Name(), nil)
+
+	absoluteLocatorStr := "github.com/kurtosis-tech/ethereum-package/src/package_io/input_parser.star"
+	commitHash := "da55be84861e93ce777076e545abee35ff2d51ce"
+
+	absoluteLocatorWithCommit := startosis_packages.NewPackageAbsoluteLocator(absoluteLocatorStr, commitHash)
```

Comment:
It's a bit unclear to me what the behavior was prior to enabling this. Did we not handle retrieving git repos from off of branches, commits, and hashes? I thought we had that actually. 

Maybe we can make this more explicit by adjusting this test to show the behavior for an absolute locator with no specified tag, branch, or commit, and one with?

I'd imagine we could do this by comparing that the contents from each are different - an requiring they equal what we expect? What do you think?

Reply:
We already had support for retrieving packages from different tag, branch or commit but only for `packageId` values and not for `absolute locators`

I'm adding that test, good.